hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleResult.java

```diff
@@ -16,76 +16,122 @@
- * errors, are captured in the {@link PreHandleResult}. This is then made available to the transaction
- * during the "handle" phase as part of the HandleContext.
+ * Metadata collected when transactions are handled as part of "pre-handle". This is then made available to the
+ * handle workflow.
  *
- * @param txnBody Transaction that is being pre-handled
- * @param payer payer for the transaction
- * @param status {@link ResponseCodeEnum} status of the transaction
- * @param payerKey payer key required to sign the transaction. It is null if payer is missing
+ * @param payer payer for the transaction, which could be from the transaction body, or could be a node account.
+ *              The payer **may be null** only in the event of a catastrophic unknown exception. That is, if the
+ *              status is {@link Status#UNKNOWN_FAILURE}, then the payer will be null. If the status is
+ *              {@link Status#NODE_DUE_DILIGENCE_FAILURE}, then the payer will be the node account. In all other cases,
+ *              the payer is extracted from the transaction body.
+ * @param status {@link Status} of this pre-handle. Will always be set.
+ * @param responseCode {@link ResponseCodeEnum} to the transaction as determined during pre-handle. Will always be set.
+ * @param txInfo Information about the transaction that is being handled. If the transaction was not parseable, then
+ *               this will be null, and an appropriate error status will be set.
+ * @param payerVerification A {@link Future} yielding the {@link SignatureVerification} for the payer.
+ * @param nonPayerVerifications A map from non-payer {@link Key} to the {@link Future} yielding the
+ *                              {@link SignatureVerification} for that key.
+ * @param nonPayerHollowVerifications A map from {@link AccountID#accountNum()} to the {@link Future} yielding the
+ *                              {@link SignatureVerification} for the hollow account of that ID.
  * @param innerResult {@link PreHandleResult} of the inner transaction (where appropriate)
  */
 public record PreHandleResult(
-        @Nullable TransactionBody txnBody,
-        @Nullable SignatureMap signatureMap,
         @Nullable AccountID payer,
-        @NonNull ResponseCodeEnum status,
-        @Nullable HederaKey payerKey,
-        @NonNull Set<HederaKey> otherPartyKeys,
-        @Nullable List<TransactionSignature> cryptoSignatures,
+        @NonNull Status status,
+        @NonNull ResponseCodeEnum responseCode,
+        @Nullable TransactionInfo txInfo,
+        @Nullable Future<SignatureVerification> payerVerification,
+        @Nullable Map<Key, Future<SignatureVerification>> nonPayerVerifications,
+        @Nullable Map<Long, Future<SignatureVerification>> nonPayerHollowVerifications,
         @Nullable PreHandleResult innerResult) {
 
+    /**
+     * An enumeration of all possible types of pre-handle results.
+     */
+    public enum Status {
```

Comment:
Are we adding this Status in addition to `ResponseCodeEnum` so we can easily check in `handle` and throw away if its `PRE_HANDLE_FAILURE` ?